packages/boxed-expression-component/src/selection/BeeTableSelectionContext.tsx

```diff
@@ -811,37 +827,78 @@ export function BeeTableSelectionContextProvider({ children }: React.PropsWithCh
     const active = selection.active;
 
     const { startRow, endRow, startColumn, endColumn } = getSelectionIterationBoundaries(selection);
+
+    const maxCellCount = currentRefs.get(0)?.size ?? 0;
+
+    Array.from(currentRefs.keys()).forEach((rowIndex) => {
+      for (let c = 0; c < maxCellCount; c++) {
+        currentRefs
+          .get(rowIndex)
+          ?.get(c)
+          ?.forEach((e) => e.setStatus?.({ isActive: false, isEditing: false, isSelected: false }));
+      }
+    });
+
     for (let r = startRow; r <= endRow; r++) {
+      const reindexedUpperRow: Array<number> = new Array<number>();
+      let reIndexUpper = 0;
+
+      if (r + 1 < 0) {
+        for (let colIndex = 0; colIndex < maxCellCount; colIndex++) {
+          if (currentRefs.get(r)?.get(colIndex)) {
+            reindexedUpperRow.push(reIndexUpper);
+            reIndexUpper += getCellColSpan(currentRefs.get(r)?.get(colIndex));
+          }
+        }
+      }
```

Comment:
lets say original `currentRefs` contain column indexes
`row[-2]: [0, 1, 2, 3] `
`row[-1]: [0, 1, 2, 3, 4, 5]`

Furthermore, the thirdcell in first row span two columns, so real indexes should be:
`row[-2]: [0, 1, 2, 4, 5] `
`row[-1]: [0, 1, 2, 3, 4, 5]`

In the DMN boxed expression editor it is even more complicated, because for input columns and annotation columns the second row (`row[-1]`) doesn't contain cells at all, so the 

original `currentRefs` are like
`row[-2]: [0, 1, 2, 3] `
`row[-1]: [0, 2, 3]`

and reindexed currentrews should be like
`row[-2]: [0, 1, 2, 4, 5] `
`row[-1]: [0, 2, 3]`